ftdetect/gofiletype.vim

```diff
@@ -31,4 +31,8 @@ au BufReadPost *.s call s:gofiletype_post()
 
 au BufRead,BufNewFile *.tmpl set filetype=gohtmltmpl
 
+au BufNewFile *.mod setfiletype gomod | setlocal fileencoding=utf-8 fileformat=unix
+au BufRead *.mod call s:gofiletype_pre("gomod")
+au BufReadPost *.mod call s:gofiletype_post()
```

Comment:
This conflicts with a built-in filetype: https://github.com/vim/vim/blob/master/runtime/filetype.vim#L996-L1002

Would probably be better to explicitly use `go.mod`, and maybe even check if the first line matches `module <ImportPath>`?

Reply:
Good catch @Carpetsmoker !

`go.mod` seems like the right call to me, maybe _in conjunction_ with check the first line?

Reply:
I'll add the checking line 👍  However, can I really use a `.` inside a ft? I checked the link martin gave and a filetype is always a single word. I think we keep it `gomod`. It's not that bad.